1_data/src/subset_nwm.R

```diff
@@ -99,8 +99,28 @@ subset_nwm <- function(ind_file, model_configuration, comids, remake_file, gd_co
     }
   }
 
+  # need to loop through the sites because the dimensions in streamflow matrix were written incorrectly
+  for(s in 1:length(site_inds)) {
+    if(length(dimids) == 2) {
+      # Note axis order is assumed here!!!
+      ncvar_put(new_nc,
+                new_nc$var$streamflow,
+                streamflow[,s]*(1/new_nc$var$streamflow$scaleFact),  # need to multiply streamflow data by scale factor in nc var
+                start = c(s, 1),
+                count = c(1,-1))
+    } else if(length(dimids) == 3) {
+      for(r in 1:nc$dim$reference_time$len) {
+        ncvar_put(new_nc,
+                  new_nc$var$streamflow,
+                  streamflow[,s]*(1/new_nc$var$streamflow$scaleFact),  # need to multiply streamflow data by scale factor in nc var
+                  start = c(s, 1, r),
+                  count = c(1,-1, 1))
+      }
+    }
+  }
+
```

Comment:
I'm implementing this a little more directly. We don't really need the streamflow array at all.

Reply:
:+1:
and thanks for giving up your vacation time for this! it really is OK if you need to wait until next week, but we'll gladly use whatever you're able to work up sooner

Reply:
With what I'm implementing locally, I now get the following for nc (from thredds) and new_nc (from a local file):
`plot(ncvar_get(nc, "streamflow",start=c(site_inds[1],1), count=c(1,-1)))`
<img width="834" alt="screen shot 2018-05-09 at 3 51 45 pm" src="https://user-images.githubusercontent.com/1492803/39836316-ef2e739a-53a0-11e8-8c49-bc9ad3883611.png">
`plot(ncvar_get(new_nc, "streamflow",start=c(1,1), count=c(1,-1)))`
<img width="830" alt="screen shot 2018-05-09 at 3 51 22 pm" src="https://user-images.githubusercontent.com/1492803/39836349-05e7bc2c-53a1-11e8-9775-d2cc4d3596e4.png">

Reply:
looks great!

Reply:
To be thorough... 
`plot(ncvar_get(nc,"streamflow",start=c(site_inds[1],1,1), count=c(1,-1,1)))`
<img width="830" alt="screen shot 2018-05-09 at 4 02 59 pm" src="https://user-images.githubusercontent.com/1492803/39836812-9c23b2bc-53a2-11e8-8c38-517de2649580.png">

`plot(ncvar_get(new_nc,"streamflow",start=c(1,1,1), count=c(1,-1,1)))`
<img width="833" alt="screen shot 2018-05-09 at 4 03 25 pm" src="https://user-images.githubusercontent.com/1492803/39836818-a3350902-53a2-11e8-8f56-c46f4e862c9f.png">

Reply:
awesome!

Reply:
Should have done this the first time around... sorry, I was in a rush and not going through the right checks. I'm rebuilding all the netcdf files now and I think you should have usable output.